teachertool/src/services/backendRequests.ts

```diff
@@ -92,29 +91,15 @@ export async function loadTestableCollectionFromDocsAsync<T>(fileNames: string[]
 }
 
 export async function askCopilotQuestionAsync(shareId: string, question: string): Promise<string | undefined> {
-    const { state: teacherTool } = stateAndDispatch();
-
-    const url = `${
-        teacherTool.copilotEndpointOverride ? teacherTool.copilotEndpointOverride : pxt.Cloud.apiRoot
-    }copilot/question`;
-
+    const url = `/api/copilot/question`;
     const data = { id: shareId, question };
     let result: string = "";
     try {
-        const headers = await auth.getAuthHeadersAsync();
-        headers["Content-Type"] = "application/json";
-
-        const request = await fetch(url, {
-            method: "POST",
-            headers,
-            body: JSON.stringify(data),
-            credentials: "include",
-        });
-
-        if (!request.ok) {
-            throw new Error(Strings.UnableToReachAI);
+        const request = await pxt.auth.AuthClient.staticApiAsync(url, data, "POST");
+        if (request.statusCode != 200) {
```

Comment:
```suggestion
        if (request.statusCode !== 200) {
```
This might be fine in this scenario, but I always give pause when I see the single equals in JS.

Reply:
Some days I just really miss C# 🙂

Reply:
I'm surprised we don't have an `isHttpSuccess(code)` util. You could do something like `Math.floor(request.statusCode / 100) === 2`. :)

Reply:
Just went to double check there wasn't one and saw the object returned by staticApiAsync has a success field. I'll switch to that instead.

Reply:
It looks like `staticApiAsync` actually does that check, @eanders-ms.. So we could technically also just use a check to see if the request was a success or not? I think this check is fine, but just wanted to put that out there. 

https://github.com/microsoft/pxt/blob/26c79ec926881e3b378008c0d37c1aa4925436bc/pxtlib/auth.ts#L616